html/includes/modal/edit_alert_transport.inc.php

```diff
@@ -0,0 +1,246 @@
+                        <input type="hidden" name="transport_id" id="transport_id" value="">
+                        <input type="hidden" name="type" id="type" value="alert-transports">
+                        <div class='form-group' title="The description of this alert transport.">
+                            <label for='name' class='col-sm-3 col-md-2 control-label'>Transport name: </label>
+                            <div class='col-sm-9 col-md-10'>
+                                <input type='text' id='name' name='name' class='form-control validation' maxlength='200' required>
+                            </div>
+                        </div>
+                        <div class="form-group" title="The type of transport.">
+                            <label for='transport-choice' class='col-sm-3 col-md-2 control-label'>Transport type: </label>
+                            <div class="col-sm-3">
+                                <select name='transport-choice' id='transport-choice' class='form-control'>
+                                    <option value="mail-form" selected>Mail</option>
+                                    <option value="ciscospark-form">Cisco Spark</option>
+                                    <!--Insert more transport type options here has support is added. Value should be: [transport_name]-form -->
+                                </select>
+                            </div>
+                        </div>
+                        <div class="form-group" title="The transport is default.">
+                            <label for="default" class="col-sm-3 col-md-2 control-label">Default Alert: </label>
+                            <div class="col-sm-2">
+                                <input type="checkbox" name="is_default" id="is_default">
+                            </div>
+                        </div>
+                    </form>
+<?php
+$transports = Config::get('alert.transports');
+// Dynamically create transport forms
+foreach (array_keys($transports) as $transport) {
+    $class = 'LibreNMS\\Alert\\Transport\\'.ucfirst($transport);
+    
+    if (!method_exists($class, 'configTemplate')) {
+        // Skip to next transport since support has not been added
+        continue;
+    }
+    
+    echo '<form method="post" role="form" id="'.strtolower($transport).'-form" class="form-horizontal transport">';
+    echo '<input type="hidden" name="transport-type" id="transport-type" value="'.strtolower($transport).'">';
+   
+    $tmp = call_user_func($class.'::configTemplate');
+
+    foreach ($tmp as $item) {
+        echo '<div class="form-group" title="'.$item['descr'].'">';
+        echo '<label for="'.$item['name'].'" class="col-sm-3 col-md-2 control-label">'.$item['title'].': </label>';
+        echo '<div class="col-sm-9 col-md-10">';
+        echo '<input type="'.$item['type'].'" id="'.$item['name'].'" name="'.$item['name'].'" class="form-control" ';
```

Comment:
We need to support checkboxes and selects here as well.

Reply:
Yes for the returned transport template, for type we could have options: text, checkbox, select etc. And the html code would have an if/else or case statement to deal with how it shows up